lib/web_ui/test/engine/semantics/semantics_test.dart

```diff
@@ -665,6 +665,29 @@ void _testHeader() {
     semantics().updateSemantics(builder.build());
     expectSemanticsTree('''
 <sem role="group" aria-label="Header of the page" style="$rootSemanticStyle"><sem-c><sem></sem></sem-c></sem>
+''');
+
+    semantics().semanticsEnabled = false;
+  });
+
+  test('renders aria-level tag for headers with heading level', () {
+    semantics()
+      ..debugOverrideTimestampFunction(() => _testTime)
+      ..semanticsEnabled = true;
+
+    final ui.SemanticsUpdateBuilder builder = ui.SemanticsUpdateBuilder();
+    updateNode(
+      builder,
+      flags: 0 | ui.SemanticsFlag.isHeader.index,
+      headingLevel: 1,
+      label: 'Header of the page',
+      transform: Matrix4.identity().toFloat64(),
+      rect: const ui.Rect.fromLTRB(0, 0, 100, 50),
+    );
```

Comment:
Let's also test that we can update it and clear `aria-level` when it's gone (assuming having the `heading` role without `aria-level` is a meaningful state).

Reply:
Hey @yjbanov, I'd say i's not a meaningful state, aria-level is required for "heading" role. Do you think I still should add it?.

Reply:
Nope, if it's not an expected state, we don't need to test it.